fiftyone/factory/repos/delegated_operation.py

```diff
@@ -167,6 +173,13 @@ def _create_indexes(self):
                 )
             )
 
+        if "dataset_id_1" not in index_names:
+            indices_to_create.append(
+                IndexModel(
+                    [("dataset_id", pymongo.ASCENDING)], name="dataset_id_1"
+                )
+            )
+
```

Comment:
might be good to have a compound index on dataset id then other attributes we want to filter/sort by. but this is required at minimum.